internal/interpreter/brainfuck.go

```
@@ -0,0 +1,27 @@
+package interpreter
+
+import "fmt"
+
+func Interpret(program string)  {
+	operations := findCommands(program)
+
+	m := newMemory(memorySize)
+
+	for _, o := range operations {
+		o.Execute(m)
+	}
+	fmt.Println()
+}
+
+func findCommands(program string) []operation {
+	var stack = &[][]operation{{}}
+	for i, symbol := range program {
+		if o, ok := commands[symbol]; ok {
+			o(stack)
+		} else {
+			panic(fmt.Sprintf("unexpected symbol in brainfuck %s in %d place", string(symbol), i+1))
```

why panic? you can just return an error in this case.

---

but why?
I think it's like unexpected operation in input program and like in Go, if we write, for example, fun instead of func go compile will argue and program will stop

---

agreed. pls write about that in the docs. A user should know about that limitation. Also pls add a test for that scenario.